asapdiscovery-data/asapdiscovery/data/schema_v2/ligand.py

```diff
@@ -0,0 +1,240 @@
+        description="ExperimentalCompoundData Schema for experimental data associated with the compound",
+    )
+
+    tags: dict[str, str] = Field({}, description="Dictionary of SD tags")
+
+    data: str = Field(
+        "",
+        description="SDF file stored as a string to hold internal data state",
+        repr=False,
+    )
+    data_format: DataStorageType = Field(
+        DataStorageType.sdf,
+        description="Enum describing the data storage method",
+        allow_mutation=False,
+    )
+
+    @validator("tags")
+    @classmethod
+    def _validate_tags(cls, v):
+        # check that tags are not reserved attribute names
+        reser_attr_names = cls.__fields__.keys()
+        for k in v.keys():
+            if k in reser_attr_names:
+                raise ValueError(f"Tag name {k} is a reserved attribute name")
+        return v
+
+    @classmethod
+    def from_oemol(
+        cls, mol: oechem.OEMol, compound_name: Optional[str] = None, **kwargs
+    ) -> "Ligand":
+        sdf_str = oemol_to_sdf_string(mol)
+        return cls(data=sdf_str, compound_name=compound_name, **kwargs)
+
+    def to_oemol(self) -> oechem.OEMol:
+        mol = sdf_string_to_oemol(self.data)
+        return mol
+
+    @classmethod
+    def from_smiles(
+        cls, smiles: str, compound_name: Optional[str] = None, **kwargs
+    ) -> "Ligand":
+        mol = smiles_to_oemol(smiles)
+        sdf_str = oemol_to_sdf_string(mol)
+        return cls(data=sdf_str, compound_name=compound_name, **kwargs)
+
+    @property
+    def smiles(self) -> str:
+        mol = sdf_string_to_oemol(self.data)
+        return oemol_to_smiles(mol)
+
+    @property
+    def inchi(self) -> str:
+        mol = sdf_string_to_oemol(self.data)
+        return oemol_to_inchi(mol)
+
+    @property
+    def inchikey(self) -> str:
+        mol = sdf_string_to_oemol(self.data)
+        return oemol_to_inchikey(mol)
+
+    @classmethod
+    def from_sdf(
+        cls,
+        sdf_file: Union[str, Path],
+        compound_name: Optional[str] = None,
+        read_SD_attrs: bool = True,
+        **kwargs,
+    ) -> "Ligand":
+        """
+        Read in a ligand from an SDF file.
+        If read_SD_attrs is True, then SD tags will be read in as attributes, overriding kwargs where double defined.
+        If read_SD_attrs is False, then SD tags will not be read in as attributes, and kwargs will be used instead
+
+        Parameters
+        ----------
+        sdf_file : Union[str, Path]
+            Path to the SDF file
+        compound_name : Optional[str], optional
+            Name of the compound, by default None
+        read_SD_attrs : bool, optional
+            Whether to read in SD tags as attributes, by default True, overrides kwargs
+        """
+        # directly read in data
+        sdf_str = read_file_directly(sdf_file)
+        lig = cls(data=sdf_str, compound_name=compound_name, **kwargs)
+        if read_SD_attrs:
+            lig.pop_attrs_from_SD_data()
```

Comment:
this method doesn't actually pop anything, right? I could definitely be wrong about this, but in my mind "pop" means "remove from source and return"

Reply:
Does now as discussed :)